lib/container.ex

```diff
@@ -151,6 +152,25 @@ defmodule Testcontainers.Container do
     %__MODULE__{config | auth: registry_auth_token}
   end
 
+  @doc """
+  Set the regular expression to check the image validity.
+
+  When using a string, it will compile it to a regular expression matching the beginning and end boundary of the word. If the compilation fails, it will match any string.
+  """
+  def with_check_image(%__MODULE__{} = config, check_image) when is_binary(check_image) do
+    case Regex.compile(".*\\b#{check_image}\\b.*") do
+      {:ok, regex} ->
+        with_check_image(config, regex)
+
+      _ ->
+        config
```

Comment:
I would rather have it crash, or, log the compilation error and explicitly set the regex to wildcard with `with_check_image(config, ~r/.*/)`. But I lean towards the first.

Reply:
I agree; it's better to let the end user know something went wrong than to try to be clever.

Reply:
I solved it on commit c447aba377b98ae3ca7267a5f3921479890f129e. I'm unsure if I should let the `Regex` raise its error or raise an `ArgumentError` with a custom message.